### PR TITLE
Fix win-32 hashing bug (from Stuart Archibald)

### PR DESCRIPTION
**Pull request by mingwandroid**

### CODE REVIEW

**Comment by mingwandroid** [comment]
With this change, the Anaconda Distro CI passes green on all builds. CC @gmarkall @stuartarchibald

**Comment by stuartarchibald** [comment]
xref: Closes https://github.com/numba/numba/issues/5179

@mingwandroid Thanks for following up on the issue, patching and testing! Much appreciated.

xref: https://github.com/numba/numba/pull/6250#issuecomment-693529389 my guess at the patch!

**Comment by mingwandroid** [comment]
Should be good to go now. Thanks all.

**Review by stuartarchibald — APPROVED** [review]
Thanks again @mingwandroid!

File: numba/cpython/hashing.py [hunk 1]
```diff
@@ -264,7 +264,7 @@ def _PyHASH_XXROTATE(x):
         @register_jitable(locals={'x': types.uint64})
         def _PyHASH_XXROTATE(x):
             # Rotate left 13 bits
-            return ((x << types.uint64(13)) | (x >> types.uint64(16)))
+            return ((x << types.uint64(13)) | (x >> types.uint64(19)))
```
Comment (stuartarchibald) [inline numba/cpython/hashing.py:267]: For reference, this line https://github.com/python/cpython/blob/b738237d6792acba85b1f6e6c8993a812c7fd815/Objects/tupleobject.c#L362 is the source of these constants.

**Pull request merged**